CRM/Civicase/Helper/CaseUrl.php

```diff
@@ -27,19 +27,44 @@ public static function getDetailsPage($caseId) {
         'status_id',
       ],
     ]);
-    $caseCategoryName = CaseCategoryHelper::getCaseCategoryNameFromOptionValue(
-      $case['case_type_id.case_type_category']
-    );
     $caseFilters = [
       'case_type_id' => [$case['case_type_id.name']],
       'status_id' => [$case['status_id']],
     ];
-    $caseDetailsUrlPath = 'civicrm/case/a/?case_type_category=' . $caseCategoryName
-      . '#/case/list?caseId=' . $caseId
-      . '&sf=id&sd=DESC&cf='
-      . urlencode(json_encode($caseFilters));
+    $caseDetailsUrlPath = sprintf(
+      'civicrm/case/a/?case_type_category=%d#/case/list?caseId=%d&sf=id&sd=DESC&cf=%s',
+      $case['case_type_id.case_type_category'],
+      $caseId,
+      urlencode(json_encode($caseFilters))
+    );
 
     return CRM_Utils_System::url($caseDetailsUrlPath, NULL, TRUE);
   }
 
+  /**
+   * Get the URL of the route type specified, for Cases category.
+   *
+   * @return string
+   *   Url to be returned.
+   */
+  public static function getUrlByRouteType(string $routeType) {
+    $categoryId = civicrm_api3('OptionValue', 'getsingle', [
+      'option_group_id' => 'case_type_categories',
+      'name' => CaseCategoryHelper::CASE_TYPE_CATEGORY_NAME,
+      'return' => ['value'],
+    ])['value'];
+
+    if ($routeType == 'dashboard') {
+      return "civicrm/case/a/?p=dd#/case?case_type_category={$categoryId}";
+    }
+    if ($routeType == 'all') {
+      return 'civicrm/case/a/?p=mg#/case/list?cf={"case_type_category":"' . $categoryId . '"}';
+    }
+    if ($routeType == 'manage_workflows') {
+      return 'civicrm/workflow/a?case_type_category=' . $categoryId . '&p=al#/list';
+    }
+
```

Comment:
Instead of having the menu URL's defined in two places and if there isa change in the manage_workflows URL, we would have to change here and in the CaseCategoryMenu service also.

Why don't we do this:
We can use the `Service/CaseCategoryMenu::getSubmenus(CaseCategoryHelper::CASE_TYPE_CATEGORY_NAME)`
Then we can do something like this
```php
$submenus = array_column($submenus, 'url', 'name');
  if ($routeType == 'dashboard') {
      return $submenus["{$caseTypeCategoryName}_dashboard"];
    }
    if ($routeType == 'all') {
      return $submenus["all_{$caseTypeCategoryName}"];
    }
    if ($routeType == 'manage_workflows') {
      return $submenus["manage_{$caseTypeCategoryName}_workflows",];
    }
```

Reply:
Looks very good your idea. Implemented here: https://github.com/compucorp/uk.co.compucorp.civicase/pull/756/commits/b7d369b562d78ba7bcf382209f6a06251b78ac8e